Added DATE argument to denote-fz-add-signature

phmcc opened the pull request:
The denote-rename-file function requires a fifth DATE argument to work correctly.  This change pulls the identifier date from the current file and feeds it to the function to satisfy this requirement.

phmcc committed:
Added DATE argument to denote-fz-add-signature

The denote-rename-file function requires a fifth DATE argument to work correctly.  This change pulls the identifier date from the current file and feeds it to the function to satisfy this requirement.

MirkoHernandez commented:
The DATE parameter is optional, it uses the current note's date value if nil. This is the behavior of the current `denote-rename-file`  command so I don't think the pr is necessary, maybe previous versions of denote have a different behavior, what version of denote are you using?

phmcc commented:
I'm currently using v3.1.0, which is what's on GNU ELPA.

For what it's worth, before implementing this fix, I was getting a `Wrong number of arguments: (5 . 5), 4` error whenever I invoked `denote-fz-add-signature` from within a `denote` buffer.   When I added a fifth `nil` argument to `denote-rename-file`, it would update the timestamp/identifier to the last modified time (thus changing the front matter and the name of the file, potentially breaking links).  This code is my way to ensure that the original timestamp persists whenever a `denote-fz` signature is added.

MirkoHernandez commented:
In the `denote-rename-file` documentation it is stated that the identifier should not be changed unless it is missing. I'm unable to reproduce the issue, could you try reproducing the error using `toggle-debug-on-error` ?, maybe the error is not related to `denote-rename-file`.

phmcc commented:
I guess it's possible.  Either way, here's the output of `toggle-debug-on-error`:

```
Debugger entered--Lisp error: (wrong-number-of-arguments (5 . 5) 4)
  denote-rename-file("/home/matogoro/Documents/Zettelkasten/Notes/Topic/20241222T172411==unnumbered--test-note__emacs.org" "Test Note" nil "1a")
  denote-fz-add-signature()
  funcall-interactively(denote-fz-add-signature)
  command-execute(denote-fz-add-signature record)
  execute-extended-command(nil "denote-fz-add-signature" nil)
  funcall-interactively(execute-extended-command nil "denote-fz-add-signature" nil)
  command-execute(execute-extended-command)
```

Let me know what you think.  Theoretically, it could be something that's piped into `denote-rename-file` rather than `denote-rename-file` itself.

MirkoHernandez commented:
I pushed some changes,  could you try the latest commit?  I'm using your exact denote version (3.1.0) and I'm unable to reproduce the issue. I had some uncommited code while testing but it does not seem related. Similar to your  example I usually have some unnumbered notes that I rename (in dired).  It seems to me that the only explanation would be a call to a different, older version of `denote-rename-file`.

phmcc commented:
Hm... I updated to your latest version (v0.3.0) but I still have the problem.  Same traceback as before, tried in both the note buffer itself and in the dired buffer.

One thought is that I've modified my setup so that `denote-fz-replace-dired-mode` is `nil`, although I just tried enabling it and is still doesn't work (both from the note buffer and from `dired`).

Does adding in the fifth argument necessarily break anything on your end?

MirkoHernandez commented:
I think I found the problem, some of the latest changes to denote (I'm probably using the same version but you are using a different commit) use a different version of `denote-rename-file`. This latest iteration does not have optional parameters and is causing the error. Thanks a lot @matogoro,  this pr will help prevent similar issues.
